system_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package systems_test
+
+import (
+	"encoding/json"
+
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/systems"
+	"github.com/juju/systems/channel"
+)
+
+type systemSuite struct {
+	testing.CleanupSuite
+}
+
+var _ = gc.Suite(&systemSuite{})
```

Comment:
It seems some tests are missing to catch the issues raised above

eg we have "system#os=ubuntu" but there's no test for a full system string